backend/api/Areas/Insurances/Controllers/InsuranceController.cs

```diff
@@ -0,0 +1,85 @@
+using System.Collections.Generic;
+using MapsterMapper;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Pims.Api.Areas.Insurance.Models;
+using Pims.Api.Policies;
+using Pims.Dal;
+using Pims.Dal.Security;
+using Swashbuckle.AspNetCore.Annotations;
+
+namespace Pims.Api.Areas.Insurance.Controllers
+{
+    /// <summary>
+    /// InsuranceController class, provides endpoints for interacting with insurances.
+    /// </summary>
+    [Authorize]
+    [ApiController]
+    [ApiVersion("1.0")]
+    [Area("insurances")]
+    [Route("v{version:apiVersion}/leases/{leaseId}/[area]")]
+    [Route("/leases/{leaseId}/[area]")]
+    public class InsuranceController : ControllerBase
+    {
+        #region Variables
+        private readonly IPimsService _pimsService;
+        private readonly IMapper _mapper;
+        #endregion
+
+        #region Constructors
+        /// <summary>
+        /// Creates a new instance of a InsuranceController class, initializes it with the specified arguments.
+        /// </summary>
+        /// <param name="pimsService"></param>
+        /// <param name="mapper"></param>
+        ///
+        public InsuranceController(IPimsService pimsService, IMapper mapper)
+        {
+            _pimsService = pimsService;
+            _mapper = mapper;
+        }
+        #endregion
+
+        #region Endpoints
+        /// <summary>
+        /// Updates a list of insurances for an insurance.
+        /// </summary>
+        /// <returns></returns>
+        [HttpPost()]
+        [HasPermission(Permissions.LeaseEdit)]
+        [Produces("application/json")]
+        [ProducesResponseType(typeof(Api.Models.BatchUpdate.Reply<InsuranceModel>), 200)]
+        [SwaggerOperation(Tags = new[] { "insurance" })]
+        public IActionResult BatchUpdate(int leaseId, Api.Models.BatchUpdate.Request<InsuranceModel> batchUpdate)
+        {
+            IList<Dal.Entities.PimsInsurance> updatedEntities = new List<Dal.Entities.PimsInsurance>();
+            foreach (Api.Models.BatchUpdate.EntryModification<InsuranceModel> modification in batchUpdate.Payload)
+            {
+                Dal.Entities.PimsInsurance insuranceEntity = _mapper.Map<Dal.Entities.PimsInsurance>(modification.Entry);
+                insuranceEntity.LeaseId = leaseId;
+                if (modification.Operation == Constants.UpdateType.Update)
+                {
+                    var insurance = _pimsService.Insurance.Update(insuranceEntity, false);
+                    updatedEntities.Add(insurance);
+                }
```

Comment:
I don't know where we landed on a decision about this "batch" update pattern but in principle I don't like that someone external to the API (ie the frontend) is dictating what entities should be added vs updated vs deleted (via that operation type)

How is this better than having the backend decide what it needs to do? ie

```cs
public static void InsertUpdateOrDeleteGraph(BloggingContext context, Blog blog)
{
    var existingBlog = context.Blogs
        .Include(b => b.Posts)
        .FirstOrDefault(b => b.BlogId == blog.BlogId);

    if (existingBlog == null)
    {
        context.Add(blog);
    }
    else
    {
        context.Entry(existingBlog).CurrentValues.SetValues(blog);
        foreach (var post in blog.Posts)
        {
            var existingPost = existingBlog.Posts
                .FirstOrDefault(p => p.PostId == post.PostId);

            if (existingPost == null)
            {
                existingBlog.Posts.Add(post);
            }
            else
            {
                context.Entry(existingPost).CurrentValues.SetValues(post);
            }
        }

        foreach (var post in existingBlog.Posts)
        {
            if (!blog.Posts.Any(p => p.PostId == post.PostId))
            {
                context.Remove(post);
            }
        }
    }

    context.SaveChanges();
}
```

Reply:
That is an interesting option. I think we have a meeting to figure out what pattern we like better.

Reply:
I think for this PR we are allowing this pattern but we will need to agree on a pattern going forward. I will create a meeting for after the holidays to come up with a unified pattern.